src/Client.ts

```diff
@@ -0,0 +1,140 @@
+import * as dgram from 'dgram'
+import * as Interfaces from './Interfaces'
+
+export default class Client
+{
+    private _id: number;
+    private _username: string;
+    private _socket: dgram.Socket;
+    private _server: Interfaces.Address;
+
+    constructor(id: number, username: string)
+    {
+        this._id = id;
+        this._username = username;
+        this._server = {
+            ip: 'localhost',
+            port: 8000
+        };
+        this._socket = dgram.createSocket('udp4');
+    }
+
+    public connect(server?:Interfaces.Address): Promise <Interfaces.Address>
+    {
+        return new Promise<Interfaces.Address>((resolve, reject) =>
+        {
+            this._socket.bind();
+
+            this._socket.on('listening', () =>
+            {
+                console.log(`[CLIENT] Listening on port ${this._socket.address().port}`)
+            });
+
+            this._socket.on('message', (msg,rinfo) =>
+            {
+                let message: Interfaces.IMessage;
+                message = JSON.parse(msg.toString());
+
+                switch(message.type) 
+                { 
+                    case Interfaces.MessageType.MESSAGE: 
+                    { 
+                        console.log(`${message.source.username}: ${message.payload}`)
+                        break; 
+                    } 
+                    default: 
+                    { 
+                        throw new Error('Wrong message.');
+                    } 
+                 } 
+            });
+
+            if(server != null)
```

Comment:
Use `!==` instead of `!=` (see [https://en.wikipedia.org/wiki/JavaScript_syntax#Operators](https://en.wikipedia.org/wiki/JavaScript_syntax#Operators))